### PR TITLE
Durable user notifications and begin Cell embedding formalism

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -1660,6 +1662,81 @@ async def _initCoreQueues(self):
...
+
+        indxbyts = s_common.int64en(indx)
+        userbyts = s_common.uhex(useriden)
+        timebyts = s_common.int64en(mesgtime)
+        typeabrv = self.notif_abrv_type.setBytsToAbrv(mesgtype.encode())
+
+        self.notifslab.delete(userbyts + timebyts, indxbyts, db=self.notif_indx_usertime)
+        self.notifslab.delete(userbyts + typeabrv + timebyts, indxbyts, db=self.notif_indx_usertype)
+
+    async def iterUserNotifs(self, useriden):
```
Comment (invisig0th) [inline synapse/cortex.py:1729]: needs to include offset

Comment (invisig0th) [inline synapse/cortex.py:1729]: size param

File: synapse/lib/cell.py [hunk 2]
```diff
@@ -1027,9 +1028,10 @@ async def initServiceStorage(self):
         pass
 
     async def initNexusSubsystem(self):
-        mirror = self.conf.get('mirror')
-        await self.nexsroot.startup(mirror, celliden=self.iden)
-        await self.setCellActive(mirror is None)
+        if self.cellparent is None:
+            mirror = self.conf.get('mirror')
+            await self.nexsroot.startup(mirror, celliden=self.iden)
+            await self.setCellActive(mirror is None)
```
Comment (vEpiphyte) [inline synapse/lib/cell.py:1034]: We never call `self.initServicePassive()` with the current delta.

Comment (invisig0th) [reply synapse/lib/cell.py:1034]: Yep, that's def included in the ticket i created which noted that we will need to eventually figure out controlling an embedded cell's active/passive callbacks and hamstring nexsoffset manipulation (required if we ever want to expose an embedded cell via telepath ).  Future self! :)

File: synapse/tests/test_cortex.py [hunk 3]
```diff
@@ -29,6 +29,52 @@ class CortexTest(s_t_utils.SynTest):
...
+
+                retn = []
+                done = asyncio.Event()
+                async def watcher():
+                    async for item in proxy.watchAllUserNotifs():
+                        retn.append(item)
+                        done.set()
+                        return
+                core.schedCoro(watcher())
+                await asyncio.sleep(0.1)
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py:61]: if this is just to get the watcher() routine running we can just call sleep(0).

Comment (invisig0th) [reply synapse/tests/test_cortex.py:61]: Not necessarily. In the case where it's double telepath it could short circuit the reply

**Review by vEpiphyte — APPROVED** [review]

File: synapse/lib/stormtypes.py [hunk 4]
```diff
@@ -6946,6 +6946,8 @@ def _ctorUserJson(self, path=None):
     def getObjLocals(self):
         return {
             'get': self._methUserGet,
+            'tell': self._methUserTell,
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py:6949]: These are missing docs.

**Review by vEpiphyte — CHANGES_REQUESTED** [review]
Need docs for stormtypes.py update

**Review by vEpiphyte — APPROVED** [review]

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**